grpc/observability/config.proto

```diff
@@ -0,0 +1,142 @@
+// Copyright 2022 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Observability Config is used by gRPC Observability plugin to control provided
+// observability features. It contains parameters to enable/disable certain
+// features, or fine tune the verbosity.
+//
+// Note that gRPC may use this config in JSON form, not in protobuf form. This
+// proto definition is intended to help document the schema but might not
+// actually be used directly by gRPC.
+
+syntax = "proto3";
+
+package grpc.observability.config;
+
+import "google/protobuf/duration.proto";
+import "grpc/observability/logging/v1/logging.proto";
+
+option java_package = "io.grpc.observability.config";
+option java_multiple_files = true;
+option java_outer_classname = "ObservabilityCOnfig";
+option go_package = "google.golang.org/grpc/observability/config";
+
+message Sampler {
+    // RateLimitingSampler enforces the average data collection intervals
+    // between calls.
+    message RateLimitingSampler {
+        // The default sample interval is 10s.
+        google.protobuf.Duration sample_interval = 1;
+    }
+
+    // ProbabilisticSampler picks calls via straightforward probability.
+    message ProbabilisticSampler {
+        // The default tracing rate is 0.1.
+        uint32 rate_nominator = 1;
+        uint32 rate_denominator = 2;
+    }
+
+    // AlwaysSampler picks every call.
+    message AlwaysSampler {}
+
+    // NeverSampler disables sampling.
+    message NeverSampler {}
+
+    oneof sampler {
+        RateLimitingSampler rate_limiting = 1;
```

Comment:
`rate_limiting_sampler`?  Note that `oneof`s are not actually a namespace in some/all/most languages' generated code.

Reply:
What I'm thinking here is the "sampler" part is mentioned in the parent field, so I can reduce one repetition. Here is an example:

```json
{
  "logging": {
	"log_filters": [
	  {
		"pattern": "Greeter/*",
		"sampler": {
		  "rate_limiting": {}
		}
	  }
	]
  }
}
```

Reply:
For the above, a `oneof` would not result in that (I think -- I'm not sure how JSON would work for certain).  If you have a `Sampler` message & corresponding field, it would for sure.